src/tests/unit/test_solc_to_k.py

```diff
@@ -4,36 +4,95 @@
 
 import pytest
 from kevm_pyk.kevm import KEVM
-from pyk.kast.inner import KToken, KVariable
+from pyk.kast.inner import KApply, KToken, KVariable
 
-from kontrol.solc_to_k import Contract, _range_predicate
+from kontrol.solc_to_k import Contract, Input, _range_predicates
 
 from .utils import TEST_DATA_DIR
 
 if TYPE_CHECKING:
     from typing import Final
 
-    from pyk.kast.inner import KInner
-
 
 EXAMPLES_DIR: Final = TEST_DATA_DIR / 'examples'
 
-PREDICATE_DATA: list[tuple[str, KInner, str, KInner | None]] = [
-    ('bytes4', KVariable('V0_x'), 'bytes4', KEVM.range_bytes(KToken('4', 'Int'), KVariable('V0_x'))),
-    ('int128', KVariable('V0_x'), 'int128', KEVM.range_sint(128, KVariable('V0_x'))),
-    ('int24', KVariable('V0_x'), 'int24', KEVM.range_sint(24, KVariable('V0_x'))),
-    ('uint24', KVariable('V0_x'), 'uint24', KEVM.range_uint(24, KVariable('V0_x'))),
+PREDICATE_DATA: list[tuple[str, KApply, list[KApply]]] = [
+    ('bytes4', KApply('bytes4', KVariable('V0_x')), [KEVM.range_bytes(KToken('4', 'Int'), KVariable('V0_x'))]),
+    ('int128', KApply('int128', KVariable('V0_x')), [KEVM.range_sint(128, KVariable('V0_x'))]),
+    ('int24', KApply('int24', KVariable('V0_x')), [KEVM.range_sint(24, KVariable('V0_x'))]),
+    ('uint24', KApply('uint24', KVariable('V0_x')), [KEVM.range_uint(24, KVariable('V0_x'))]),
+    (
+        'tuple',
+        KApply(
```

Comment:
`KApply` also takes `*args`, so you can construct these data structures a bit more concisely as:

```python
KApply('abi_tuple_type', KApply(...), ..., KApply(...))
```

Reply:
Thanks, I've updated these data structures in https://github.com/runtimeverification/kontrol/pull/78/commits/2d10a2540dc3737efe4a792be884bee875b9462d.